### PR TITLE
Use less device memory when checkpointing particles

### PR DESCRIPTION
**Pull request by atmyers**
This chunks the operation per box rather than per rank so as to use less device memory.

The proposed changes:
- [ ] fix a bug or incorrect behavior in AMReX
- [ ] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [ ] include documentation in the code and/or rst files, if appropriate

### CODE REVIEW

**Comment by ax3l** [comment]
@atmyers for your Intel CI issue: I pushed a work-around to https://github.com/AMReX-Codes/amrex/pull/3239 :)

**Review by ax3l — APPROVED** [review]
Awesome, thanks a lot! :tada:

**Comment by WeiqunZhang** [comment]
The stream sync at line 188 is not needed.

File: Src/Particle/AMReX_WriteBinaryParticleData.H [hunk 1]
```diff
@@ -190,16 +190,12 @@ packIOData (Vector<int>& idata, Vector<ParticleReal>& rdata, const PC& pc, int l
...
-    typename PC::RealVector rdata_d(rdata.size());
-
     typename PC::IntVector write_int_comp_d(write_int_comp.size());
     typename PC::IntVector write_real_comp_d(write_real_comp.size());
     Gpu::copyAsync(Gpu::hostToDevice, write_int_comp.begin(), write_int_comp.end(),
                    write_int_comp_d.begin());
     Gpu::copyAsync(Gpu::hostToDevice, write_real_comp.begin(), write_real_comp.end(),
-                   write_real_comp_d.begin());
-    Gpu::Device::streamSynchronize();
+                   write_real_comp_d.begin());\
```
Comment (WeiqunZhang) [inline Src/Particle/AMReX_WriteBinaryParticleData.H]: Why is there a `\`?

Comment (atmyers) [inline Src/Particle/AMReX_WriteBinaryParticleData.H]:
```suggestion
                   write_real_comp_d.begin());
```

**Review by WeiqunZhang — APPROVED** [review]

**Pull request merged**